app/views/partials/foyer/ressources/year-moins-2.html

```diff
@@ -4,42 +4,57 @@
     Ces informations se trouvent sur votre déclaration de revenus d'impôts {{ yearMoins2 }}.
     <br>Vous pouvez la retrouver en ligne sur <a href="http://www.impots.gouv.fr/">impots.gouv.fr</a>.
   </p>
-  <form class="form-horizontal" novalidate ng-submit="submit()">
+  <form class="form-horizontal" novalidate name="form" ng-submit="submit(form)">
     <div class="row">
       <div class="col-md-12" ng-repeat="individuRef in individuRefsToDisplay">
         <h2>{{ individuRef.label }}</h2>
-        <div class="form-group" ng-repeat="rnc in individuRef.rnc">
-          <label class="control-label col-sm-3" for="{{ rnc.categorie.id }}-{{ individuRef.individu.role }}">
+        <div
+          class="form-group"
+          ng-repeat="rnc in individuRef.rnc"
+          ng-init="inputName = rnc.categorie.id +  '-' + individuRef.individu.role"
+          ng-class="{ 'has-error': form.$submitted && form[inputName].$invalid }"
+          >
+          <label class="control-label col-sm-3" for="{{ inputName }}">
             {{ rnc.categorie.label }}
           </label>
           <div class="col-sm-3">
             <span class="input-group">
               <input
                 type="number"
-                id="{{ rnc.categorie.id }}-{{ individuRef.individu.role }}"
+                id="{{ inputName }}"
+                name="{{ inputName }}"
                 class="form-control text-right"
                 ng-model="rnc.montant"
                 min="0"
                 select-on-focus
-                zero-to-empty>
+                zero-to-empty
+                aria-describedby="help-{{ inputName }}">
               <span class="input-group-addon hidden-xs">€</span>
             </span>
+            <ul class="text-danger" ng-if="form.$submitted && form[inputName].$invalid">
```

Comment:
Using `ng-show` rather than `ng-if` would prevent the whole layout bouncing around when errors appear  🙂